daemon/daemon.go

```diff
@@ -762,10 +762,13 @@ func (h *patchConfig) Handle(params PatchConfigParams) middleware.Responder {
 	changes := d.conf.Opts.Apply(params.Configuration, changedOption, d)
 	log.Debugf("Applied %d changes", changes)
 	if changes > 0 {
-		if config.Opts.IsEnabled(endpoint.OptionPolicy) && config.EnablePolicy != endpoint.AlwaysEnforce {
-			config.EnablePolicy = endpoint.AlwaysEnforce
-		} else if !config.Opts.IsEnabled(endpoint.OptionPolicy) && config.EnablePolicy != endpoint.NeverEnforce {
-			config.EnablePolicy = endpoint.NeverEnforce
+		_, ok := params.Configuration[endpoint.OptionPolicy]
+		if ok {
+			if config.Opts.IsEnabled(endpoint.OptionPolicy) && config.EnablePolicy != endpoint.AlwaysEnforce {
+				config.EnablePolicy = endpoint.AlwaysEnforce
+			} else if !config.Opts.IsEnabled(endpoint.OptionPolicy) && config.EnablePolicy != endpoint.NeverEnforce {
```

Comment:
Before this change, this line would evaluate to 'true' even if there was no update to policy enforcement (e.g., `cilium config ConntrackLocal=true` because by default in the non-K8s case, we do not restrict traffic if there are no rules / policies added. If there was any change applied to the daemon with no policies / rules added, `!config.Opts.IsEnabled(endpoint.OptionPolicy)` would evaluate to `true` and by default, `config.EnablePolicy` is set to `config.DefaultEnforcement`. So, the daemon would then change config.EnablePolicy to `endpoint.NeverEnforce`, and the 11-getting-started.sh test would fail because policy enforcement was disabled and all traffic would be allowed.